client/login/ssr.js

```diff
@@ -33,7 +29,7 @@ export function setShouldServerSideRenderLogin( context, next ) {
  * @returns {boolean} If the value of &redirect_to= on the log-in page is compatible with SSR
  */
 function isRedirectToValidForSsr( redirectToQueryValue ) {
-	if ( 'undefined' === typeof redirectToQueryValue ) {
+	if ( redirectToQueryValue === undefined ) {
```

Comment:
Won't that trigger a `Uncaught ReferenceError: redirectToQueryValue is not defined` error?

Reply:
No, the `redirectToQueryValue` identifier is always defined because it's the function's parameter. The `typeof` check is only useful for globals like `window` that don't need to exist at all.

Reply:
Then why don't we remove the check? I just don't understand why we replace the `typeof x === 'undefined'` check with a `=== undefined`. How is the latter better?

Reply:
We need the check: if `redirectToQueryValue` is `undefined`, the result is immediately `true` because the URL doesn't have `redirectTo` param to validate. Only when it has a "real" value do we proceed to validate it.

The `=== undefined` check is simpler and sufficient. The `typeof x` check is needed only for special purposes, and in our codebase we overuse it a lot.

Reply:
In my head `typeof x === 'undefined` is much safer and I always prefer it to `=== undefined`. The only reason why one would use the latter would be because it's a bit shorter, but always prefer the safer than shorter TBH. Unless you have a better reason than brevity?

Reply:
> In my head `typeof x === 'undefined` is much safer and I always prefer it to `=== undefined`.

I think you should start using `=== undefined` 🙂 The only safety added by `typeof` is that the `typeof blah_blah` expression is valid also on identifiers that are not declared at all. But most of the time you can be 100% sure that the identifier is declared. There is a parameter declaration or variable declaration nearby.